### PR TITLE
model's $fillable attributes available in crud views

### PR DESCRIPTION
**Pull request by Frondor**
#800 so there's no need to add unnecesary fake columns if you want to, lets say, hide the Edit/Delete button if `$entry->trashed()` or `$entry->deleted_at !== null`

### CODE REVIEW

File: src/app/Http/Controllers/CrudFeatures/AjaxTable.php [hunk 1]
```diff
@@ -21,6 +21,8 @@ public function search()
                     ->pluck('name')
                     // add the primary key, otherwise the buttons won't work
                     ->merge($this->crud->model->getKeyName())
+                    // make $fillable attributes availables  in views
```
Comment (lloy0076) [inline src/app/Http/Controllers/CrudFeatures/AjaxTable.php]: available and there's two spaces after it too 👿

Comment (Frondor) [inline src/app/Http/Controllers/CrudFeatures/AjaxTable.php]: My bad, my bad!

**Comment by Frondor** [comment]
@lloy0076 well, it wont affect column/rows displaying. Since it's a merge, we're including some extra fields (the ones that weren't set manually in the controller), as part of the `$entry` object in the view.

**Comment by tabacitu** [comment]
Hi @Frondor ,

I'm afraid this has some unexpected consequences: by adding all fillable fields there, the search results will vary a lot to what has happened so far. And not for the better, IMHO. Basically the search will search within all fillable db column, not just the ones that are shown. Which seems unintuitive to me, and it might be a security issue for some.

Use case 1:
- article has **title**, excerpt, content; the ones in bold also have crud columns;
- in the past, if you searched, you searched within the titles; 4 results;
- after this PR, it would also search within excerpt and content; 44 results;

Use case 2:
- user has **name**, **email**, social_security_number; the ones in bold also have crud columns;
- in the past, if you searched, you searched within the names and emails;
- after this PR, it would also search within the social security number;

This seems unintuitive to me - sure, there are cases where you'd like the search to extend into more columns than the ones shown by CRUD, but I don't think that should be the default behaviour...

Hope you understand. Cheers!

**Comment by Frondor** [comment]
No problem, I stopped using this package long ago.

**Comment by tabacitu** [comment]
Well.. sorry to hear that. Can you let us know what you've chosen instead, and why? Maybe we have stuff to learn from them.

Thanks, cheers!

**Pull request closed without merging**